query-builder/src/main/java/com/datastax/oss/driver/internal/querybuilder/insert/DefaultInsert.java

```diff
@@ -192,6 +255,14 @@ public String asCql() {
         builder.append(timestamp);
       }
     }
+    if (ttlInSeconds != null) {
+      builder.append((timestamp != null) ? " AND " : " USING ").append("TTL ");
+      if (ttlInSeconds instanceof BindMarker) {
+        ((BindMarker) ttlInSeconds).appendTo(builder);
+      } else {
```

Comment:
That `else` is worrying me. What if someone will pass some custom object as the `Object ttlInSeconds`?
We will set it on the builder and the user will get the error while executing the query, right?
Maybe we should do 
```
else if(ttlInSeconds instanceof Integer){
...
}else{
throw IllegalArgumentException()
}
```
To fail fast on not-supported type?

Reply:
I see your point. I modeled this after the code for `timestamp`, as it supports a literal value and a BindMarker as well. For `ttlInSeconds`, I believe the literal can be an `Integer` or a `String` in this case, so long as it is a whole number. It could be a `Long` as well, as long as the value isn't larger than what is supported in Cassandra. In fact, it could be any Object that has a `toString()` implementation that returns a whole number between 0 and whatever the max is in Cassandra. 

We could be a little permissive and only throw the Exception if we can't parse an Integer value from the Object:
```
try {
  Integer ttlValue = Integer.parseInt(ttlInSeconds.toString());
  if (ttlValue.compareTo(0) < 0) {
    throw new IllegalArgumentException("Provided ttlInSeconds value can not be negative");
  }
  // add  USING TTL clause
} catch (NumberFormatException nfe) {
  throw new IllegalArgumentException("Provided ttlInSeconds is not a valid value");
}
```
I guess there is value in failing to build the Builder just prior to trying to have the Builder build the CQL. That would be the only benefit to failing at this point.